hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoUpdateHandler.java

```diff
@@ -35,26 +55,48 @@
  * This class contains all workflow-related functionality regarding {@link HederaFunctionality#CRYPTO_UPDATE}.
  */
 @Singleton
-public class CryptoUpdateHandler implements TransactionHandler {
+public class CryptoUpdateHandler extends BaseCryptoHandler implements TransactionHandler {
 
     private final CryptoSignatureWaivers waivers;
+    private StakingValidator stakingValidator;
 
     @Inject
-    public CryptoUpdateHandler(@NonNull final CryptoSignatureWaivers waivers) {
+    public CryptoUpdateHandler(
+            @NonNull final CryptoSignatureWaivers waivers, @NonNull final StakingValidator stakingValidator) {
         this.waivers = requireNonNull(waivers, "The supplied argument 'waivers' must not be null");
+        this.stakingValidator =
+                requireNonNull(stakingValidator, "The supplied argument 'stakingValidator' must not be null");
+    }
+
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        final var op = txn.cryptoUpdateAccountOrThrow();
+        validateTruePreCheck(op.hasAccountIDToUpdate(), INVALID_ACCOUNT_ID);
+        validateFalsePreCheck(
+                op.hasProxyAccountID() && !op.proxyAccountID().equals(AccountID.DEFAULT),
+                PROXY_ACCOUNT_ID_FIELD_IS_DEPRECATED);
     }
 
     @Override
     public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
         requireNonNull(waivers);
         final var txn = context.body();
+        pureChecks(txn);
+
         final var payer = context.payer();
         final var op = txn.cryptoUpdateAccountOrThrow();
-        final var updateAccountId = op.accountIDToUpdateOrElse(AccountID.DEFAULT);
 
-        final var newAccountKeyMustSign = !waivers.isNewKeySignatureWaived(txn, payer);
+        // update account must exist. Validated in pureChecks
+        final var updateAccountId = op.accountIDToUpdateOrThrow();
+
+        // 1. When updating the treasury account 0.0.2 key, the target account must also sign the transaction
+        // 2. When updating treasury account 0.0.2, even if system admin is the payer for the transaction,
+        // the target account must sign the transaction
+        // 3. If the payer for the transaction is 0.0.2 or system admin, then no signatures are needed for the update
         final var targetAccountKeyMustSign = !waivers.isTargetAccountSignatureWaived(txn, payer);
```

Comment:
Note that waivers like these are exactly what the privileged account facility is supposed to determine.  It's fine here as a double-check, but it definitely needs to get into the facility as well.